noxfile.py

```diff
@@ -158,3 +159,37 @@ def test_with_pydantic_required_package(session: nox.Session, pydantic: str):
             f"{PREFIX_TESTS_UNIT}/parser/_pydantic/",
         ],
     )
+
+
+@nox.session()
+@nox.parametrize("pydantic", ["1.10", "2.0"])
+def test_with_boto3_and_pydantic_required_package(session: nox.Session, pydantic: str):
```

Comment:
I need to install pydantic + boto3 because we export the `DynamoDBPersistenceLayer` at [top level](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/aws_lambda_powertools/utilities/idempotency/__init__.py#L12) and it requires boto3.

Reply:
I think you meant you **only** needed boto3 when using `DynamoDBPersistenceLayer`, whereas for **this test** you also need pydantic because it'll use serializers and models.